stubs/oauthlib/oauthlib/common.pyi

```diff
@@ -1,64 +1,81 @@
 
-urlencoded: Any
+UNICODE_ASCII_CHARACTER_SET: Final[str]
+CLIENT_ID_CHARACTER_SET: Final[str]
+SANITIZE_PATTERN: Final[re.Pattern[str]]
+INVALID_HEX_PATTERN: Final[re.Pattern[str]]
+always_safe: Final[str]
+log: Logger
 
-def urldecode(query): ...
-def extract_params(raw): ...
-def generate_nonce(): ...
-def generate_timestamp(): ...
-def generate_token(length: int = 30, chars=...): ...
-def generate_signed_token(private_pem, request): ...
+def quote(s: str | bytes, safe: bytes = b"/") -> str: ...
+def unquote(s: str | bytes) -> str: ...
+def urlencode(params: list[tuple[str | bytes, str | bytes]]) -> str: ...
+def encode_params_utf8(params: list[tuple[str | bytes, str | bytes]]) -> list[tuple[bytes, bytes]]: ...
+def decode_params_utf8(params: list[tuple[str | bytes, str | bytes]]) -> list[tuple[str, str]]: ...
+
+urlencoded: Final[set[str]]
+
+def urldecode(query: str | bytes) -> list[tuple[str, str]]: ...
+def extract_params(raw: str | bytes | dict[str, str] | Iterable[tuple[str, str]]) -> list[tuple[str, str]] | None: ...
+def generate_nonce() -> str: ...
+def generate_timestamp() -> str: ...
+def generate_token(length: int = 30, chars: Sequence[str] = ...) -> str: ...
+def generate_signed_token(private_pem: str, request) -> str: ...
 def verify_signed_token(public_pem, token): ...
-def generate_client_id(length: int = 30, chars=...): ...
-def add_params_to_qs(query, params): ...
-def add_params_to_uri(uri, params, fragment: bool = False): ...
-def safe_string_equals(a, b): ...
-def to_unicode(data, encoding: str = "UTF-8"): ...
+def generate_client_id(length: int = 30, chars: Sequence[str] = ...) -> str: ...
+def add_params_to_qs(query: str, params: dict[str, str] | Iterable[tuple[str, str]]) -> str: ...
+def add_params_to_uri(uri: str, params: dict[str, str] | Iterable[tuple[str, str]], fragment: bool = False) -> str: ...
+def safe_string_equals(a: str, b: str) -> bool: ...
+@overload
+def to_unicode(data: str | bytes, encoding: str = "UTF-8") -> str: ...
+@overload
+def to_unicode(data: Mapping[_K, _V], encoding: str = "UTF-8") -> dict[str, _V]: ...
+@overload
+def to_unicode(data: _T, encoding: str = "UTF-8") -> _T: ...
 
-class CaseInsensitiveDict(dict[Any, Any]):
-    proxy: Any
-    def __init__(self, data) -> None: ...
-    def __contains__(self, k): ...
-    def __delitem__(self, k) -> None: ...
-    def __getitem__(self, k): ...
-    def get(self, k, default: Incomplete | None = None): ...
-    def __setitem__(self, k, v) -> None: ...
+class CaseInsensitiveDict(dict[str, Any]):
+    proxy: dict[str, str]
+    def __init__(self, data: dict[str, Any]) -> None: ...
+    @overload
+    def __contains__(self, k: str) -> bool: ...
+    @overload
+    def __contains__(self, k: object) -> bool: ...
+    def __delitem__(self, k: str) -> None: ...
+    def __getitem__(self, k: str): ...
+    def get(self, k: str, default: Any | None = None) -> Any | None: ...
```

Comment:
Similar here. Should ideally mimic `dict.get`, but `Incomplete` is fine for now:

```suggestion
    def get(self, k: str, default: Incomplete | None = None) -> Incomplete | None: ...
```

(Maybe it's also ok to just inherit this method from `dict`?)

Reply:
I would prefer to leave it to define `k` as `str`